CONTRIBUTING.md

```diff
@@ -1,3 +1,99 @@
-Thank you for helping the Streamlink project! Before opening a new ticket,
-please make sure that there's no similar ticket already opened for your
-issue and that it isn't a [known issue](https://streamlink.github.io/players.html#known-issues-and-workarounds).
+# Contributing to Streamlink
+
+Want to get involved? Thanks! There are plenty of ways to help!
+
+
+## Reporting issues
+
+A bug is a *demonstrable problem* that is caused by the **latest version** of the code in the repository. Good bug reports are extremely helpful - thank you!
+
+Please read the following guidelines before you [report an issue][issues]:
+
+1. **See if the issue is already known** — check the list of [known issues][known-issues].
+
+2. **Use the GitHub issue search** — check if the issue has already been reported. If it has been, please comment on the existing issue.
+
+3. **Check if the issue has been fixed** — the latest `master` or development branch may already contain a fix.
+
+4. **Isolate the demonstrable problem** — make sure that the code in the project's repository is *definitely* responsible for the issue.
+
+5. **Format your issue report** — [well formatted texts][mastering-markdown] will make it much easier for developers to understand your report.
+
+Please try to be as detailed as possible in your report too. What is your environment? What steps will reproduce the issue? What would you expect to be the outcome? All these details will help people to assess and fix any potential bugs. The [issue template][issue-template] will aid you in structuring your report when submitting a new issue. If your report includes log outputs, please provide those as text in multi-line code blocks if possible for better readability, thank you!
```

Comment:
Just a couple grammatical nitpicks on this line: `What would you expect to be the outcome` would make more sense as `What outcome do you expect` or `What would you expect the outcome to be`. Then for `If your report includes log outputs,` that makes more sense as `If your report includes log output`, so no need for the s, which then removes the need for the comma.